drivers/ethernet/eth_sam_gmac_priv.h

```diff
@@ -31,7 +31,8 @@
 /** RX descriptors count for main queue */
 #define MAIN_QUEUE_RX_DESC_COUNT CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT
 /** TX descriptors count for main queue */
-#define MAIN_QUEUE_TX_DESC_COUNT (CONFIG_NET_BUF_TX_COUNT + 1)
+#define MAIN_QUEUE_TX_DESC_COUNT (max(CONFIG_NET_BUF_RX_COUNT, \
```

Comment:
Please add comment why this formula.

Reply:
While doing that, also please update the title from "drivers: eth: gmac: correctly compute the number of descriptors" to "drivers: eth: gmac: *conservatively* compute the number of descriptors". Per, https://github.com/zephyrproject-rtos/zephyr/pull/12591#issuecomment-456035299 , this is unlikely "correct".

Reply:
Also:

> This wrong assume that 

Should be "wrongly assume".